### PR TITLE
Disable SDPA attention layer for mistral and gpt_bigode

### PR DESCRIPTION
**Pull request by jiminha**
Currently both mistral and gpt-bigcode model is failing to run since SDPA attention layer is enabled for both model by default with transformer4.37.2 + torch2.2. Until we have proper FusedSDPA support for these model, we are disabling SDPA attention so it can fall back to original attention layer.

### CODE REVIEW

**Comment by jiminha** [comment]
Per @libinta , she doesn't want this to be added per model, but want to override configuration in generic way so we don't need to modify all individual model file.

File: optimum/habana/transformers/models/modeling_all_models.py [hunk 1]
```diff
@@ -113,6 +114,46 @@ def gaudi_conv1d_forward(self, x):
...
 
 
+# Adapted from transformers.modeling_utils.PreTrainedModel._check_and_enable_sdpa
+@classmethod
+def gaudi_check_and_enable_sdpa(cls, config, hard_check_only: bool = False) -> PretrainedConfig:
+
+    #This model doesn't support SDPA in Gaudi yet, fallback to original code.
+    MODELS_ATTN_IMPLEMENTATION_EAGER = [
+        "gpt_bigcode",
+        "mistral",
```
Comment (libinta) [inline optimum/habana/transformers/models/modeling_all_models.py:124]: @vivekgoe  should we add the t5 model here too?

Comment (vivekgoe) [reply optimum/habana/transformers/models/modeling_all_models.py:124]: you mean "bart". yes please go ahead and add it.

**Review by vivekgoe — APPROVED** [review]
Please add "bart" also to list of models not using sdpa.

**Review by libinta — APPROVED** [review]

**Comment by jiminha** [comment]
> Please add "bart" also to list of models not using sdpa.

@vivekgoe  change merged before I add this. I just tested adding "bart", and remove your original change. it also works for summarization. Do you want to make the change?

**Comment by astachowiczhabana** [comment]
https://github.com/huggingface/optimum-habana/pull/771

**Pull request merged**